src/bp/nlu-core/training-pipeline.ts

```diff
@@ -510,11 +510,19 @@ const TrainOutOfScope = async (input: TrainStep, tools: Tools, progress: progres
 
 const NB_STEPS = 6 // change this if the training pipeline changes
 
-export const Trainer = async (
-  input: TrainInput,
-  tools: Tools,
-  progress?: (x: number) => void
-): Promise<TrainOutput> => {
+type Func = (...args: any[]) => any
```

Comment:
isn't it the same as the builtin typescript `Function` type ?

Reply:
Nope:

![Screen Shot 2021-02-08 at 9 16 37 AM](https://user-images.githubusercontent.com/25970722/107232071-fea08e00-69ee-11eb-853b-8ca890d63299.png)